torch/distributed/tensor/parallel/style.py

```diff
@@ -324,8 +384,166 @@ def make_output_reshard_tensor(
     Return:
         A :class:`torch.Tensor` object converted from output DTensor.
     """
+    _deprecate_warnings("make_output_reshard_tensor", "Specify output_layouts instead.")
+    return make_output_shard_1d(output, device_mesh).to_local()  # type: ignore[call-arg, attr-defined, misc]
+
+
+def _is_redistribute_necessary(
```

Comment:
nit: can be fixed in follow up prs, name is not quite consistent with the check, i.e. if "==", it's actually not necessary, maybe call it like `_needs_redistribute`

Reply:
oh good point, let me update the func name.